MVPInternshipApp.Server/Controllers/CustomerController.cs

```diff
@@ -75,24 +73,32 @@ public async Task<IActionResult> PutCustomer(int id, Customer customer)
         // POST: api/Customer
         // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
         [HttpPost]
-        public async Task<ActionResult<Customer>> PostCustomer(Customer customer)
+        public async Task<ActionResult<CustomerDto>> PostCustomer(CustomerDto customerDto)
         {
+            var customer = customerDto.ToModel();
             _context.Customers.Add(customer);
             await _context.SaveChangesAsync();
 
-            return CreatedAtAction("GetCustomer", new { id = customer.Id }, customer);
+
+            return CreatedAtAction("GetCustomer", new { id = customer.Id }, new CustomerDto(customer));
         }
 
         // DELETE: api/Customer/5
         [HttpDelete("{id}")]
         public async Task<IActionResult> DeleteCustomer(int id)
         {
-            var customer = await _context.Customers.FindAsync(id);
+            var customer = await _context.Customers.Where(c => c.Id == id)
```

Comment:
Before directly looking if the ID is in the records or not, can you please verify first if the **ID's value** is a valid content? This way, you can be able to save your resource and reduce cost.